From 06bfa9a002fda838d219700bb2b836c44b588359 Mon Sep 17 00:00:00 2001
From: Yunguo Wei <yunguo.wei@windriver.com>
Date: Sat, 5 May 2018 17:49:06 +0800
Subject: [PATCH 08/12] create boot symlink based on relative path

If /boot is a seperated boot partition, grub will set root device to
boot partition and an abs path doesn't make senese.

Signed-off-by: Yunguo Wei <yunguo.wei@windriver.com>
---
 src/boot/grub2/ostree-grub-generator | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/boot/grub2/ostree-grub-generator b/src/boot/grub2/ostree-grub-generator
index 20b07c3..025c4dd 100644
--- a/src/boot/grub2/ostree-grub-generator
+++ b/src/boot/grub2/ostree-grub-generator
@@ -89,9 +89,9 @@ populate_menu()
         fi
         menu="${menu}}\n\n"
 
-        linux_dir=`dirname ${boot_prefix}${linux}`
+        linux_dir=`dirname ${linux}`
         boots[$count]=`mktemp -d ${sysroot_dir}${boot_prefix}/boot.XXXXXXXXXX`
-        ln -sf ${linux_dir} ${boots[$count]}/boot
+        ln -sf ..${linux_dir} ${boots[$count]}/boot
         ln -sf ../..${ostree} ${boots[$count]}/ostree
         count=`expr $count + 1`        
     done
-- 
2.7.4

